astropy/coordinates/builtin_frames/itrs.py

```diff
@@ -4,26 +4,68 @@
 from astropy.utils.decorators import format_doc
 from astropy.coordinates.representation import CartesianRepresentation, CartesianDifferential
 from astropy.coordinates.baseframe import BaseCoordinateFrame, base_doc
-from astropy.coordinates.attributes import TimeAttribute
-from .utils import DEFAULT_OBSTIME
+from astropy.coordinates.attributes import (TimeAttribute,
+                                            EarthLocationAttribute)
+from .utils import DEFAULT_OBSTIME, EARTH_CENTER
 
 __all__ = ['ITRS']
 
+doc_footer = """
+    Other parameters
+    ----------------
+    obstime : `~astropy.time.Time`
+        The time at which the observation is taken.  Used for determining the
+        position of the Earth and its precession.
+    location : `~astropy.coordinates.EarthLocation`
+        The location on the Earth.  This can be specified either as an
+        `~astropy.coordinates.EarthLocation` object or as anything that can be
+        transformed to an `~astropy.coordinates.ITRS` frame. The default is the
+        centre of the Earth.
+"""
 
-@format_doc(base_doc, components="", footer="")
+
+@format_doc(base_doc, components="", footer=doc_footer)
 class ITRS(BaseCoordinateFrame):
     """
     A coordinate or frame in the International Terrestrial Reference System
     (ITRS).  This is approximately a geocentric system, although strictly it is
-    defined by a series of reference locations near the surface of the Earth.
+    defined by a series of reference locations near the surface of the Earth (the ITRF).
     For more background on the ITRS, see the references provided in the
     :ref:`astropy:astropy-coordinates-seealso` section of the documentation.
+
+    This frame also includes frames that are defined *relative* to the center of the Earth,
+    but that are offset (in both position and velocity) from the center of the Earth. You
+    may see such non-geocentric coordinates referred to as "topocentric".
+
+    Topocentric ITRS frames are convenient for observations of near Earth objects where
+    stellar aberration is not included. One can merely subtract the observing site's
+    EarthLocation geocentric ITRS coordinates from the object's geocentric ITRS coordinates,
+    put the resulting vector into a topocentric ITRS frame and then transform to
+    `~astropy.coordinates.AltAz` or `~astropy.coordinates.HADec`. The other way around is
+    to transform an observed `~astropy.coordinates.AltAz` or `~astropy.coordinates.HADec`
+    position to a topocentric ITRS frame and add the observing site's EarthLocation geocentric
+    ITRS coordinates to yield the object's geocentric ITRS coordinates.
+
+    On the other hand, direct transforms between geocentric ITRS coordinates and observed
```

Comment:
That is incorrect. As currently written, there is no change in behavior for a transform from geocentric ITRS to Observed. This would still go:
```
 ITRS(g)->CIRS(g)->ICRS->CIRS(t)->ITRS(t)->Observed,
```
 where `g` denotes geocentric and `t` denotes topocentric. Thus the difference in stellar aberration will be applied.

Reply:
I see now. Can you add an in-line comment to the lines below to point out that they may invoke the ITRS self transform, that goes through CIRS and thus includes stellar aberration. This will help future developers IMO

https://github.com/astropy/astropy/blob/c9df34396397fc01919f038978f9b4fb415be4a0/astropy/coordinates/builtin_frames/itrs_to_observed_transforms.py#L111 and https://github.com/astropy/astropy/blob/c9df34396397fc01919f038978f9b4fb415be4a0/astropy/coordinates/builtin_frames/itrs_to_observed_transforms.py#L146

Also, maybe change `direct transforms` here to:

```
On the other hand, using `transform_to` to transform geocentric ITRS coordinates 
to topocentric ITRS coordinates, or ITRS coordinates to observed 
`~astropy.coordinates.AltAz` or `~astropy.coordinates.HADec` coordinates will 
include the difference....
```

And perhaps change "If the geocentric ITRS coordinates of the object include stellar aberration at the geocenter" to "If your ITRS coordinates point to a deep space object"?

Reply:
In addition, I think it would be great to update https://github.com/astropy/astropy/blob/main/docs/coordinates/common_errors.rst#altaz-calculations-for-earth-based-objects to use the new topocentric ITRS frame, as you've done in your `test_itrs_straight_overhead` function.